server/server.js

```diff
@@ -219,3 +210,16 @@ async function disconnectDataSources (dataSources) {
     }
   }
 }
+
+function newSubScriptionServer (server, schema) {
+  // Set up the WebSocket for handling GraphQL subscriptions
+  /* eslint-disable no-new */
+  new SubscriptionServer({
```

Comment:
We need to save a reference to current subscription server, close it, and create a new one.

This PR got very big and we should better merge it ignoring this problem. 
I created a new task for that: https://issues.jboss.org/browse/AEROGEAR-7696

Reply:
@aliok Okay that sounds good. In fact this whole file needs to be refactored to some extent. For example the current schema reload feature means that every time a request comes into the server, we are initializing a new GraphQL engine. This adds significant overhead to every request. I think there is a better way to do it though.